ember-core/shared/src/main/scala/org/http4s/ember/core/Parser.scala

```diff
@@ -497,21 +510,17 @@ private[ember] object Parser {
                 codeS = new String(buffer, start, idx - start)
                 state = 2
                 start = idx + 1
+              } else if (value == lf && (idx > 0 && buffer(idx - 1) == cr)) {
+                // Even though a status line must have this pattern (rfc7230)
+                //    status-line = HTTP-version SP status-code SP reason-phrase CRLF
+                // some servers might remove the last SP if they do not implement the reason phrase
+                val codeInt = new String(buffer, start, idx - start - 1).toInt
```

Comment:
Creating this `String` would allocate a `String` object, as well as its underlying byte array, which are discarded immediately. Perhaps this should better be a parsing method that reads over the bytes from the `buffer`?